pkg/agent/run.go

```diff
@@ -200,12 +208,34 @@ func gatherData(ctx context.Context, config Config) []*api.DataReading {
 			log.Printf("Running data gatherer %s of type %s as Local, data-path override present", dgConfig.Name, dgConfig.Kind)
 		}
 
-		dg, err := dgConfig.Config.NewDataGatherer(ctx)
+		// initialize data gatherer
+		newDg, err := dgConfig.Config.NewDataGatherer(ctx)
 		if err != nil {
 			log.Fatalf("failed to instantiate %s DataGatherer: %v", kind, err)
 		}
 
-		dataGatherers[dgConfig.Name] = dg
+		dg, ok := dataGatherers[dgConfig.Name]
+		if !ok {
+			// start the data gatherers and wait for the cache sync
+			err := startAndSyncDataGather(ctx, newDg)
+			if err != nil {
+				log.Printf("failed to start and cache sync %s DataGatherer: %v", kind, err)
+			}
+			dataGatherers[dgConfig.Name] = newDg
+		} else {
+			// checking if there have been any changes to the data gatherer's config
```

Comment:
This was done to still allow changes in the data gatherer config at runtime, since it was a behaviour that we had, i tried to keep it. Not sure what is your opinion on `dg.Equals(newDg)`, I've added Equals to the interface, it dectects changes in the gatherer config fields.